azavea/raster-foundry
# Upload status from Evaporate

This PR simplifies the upload and transfer fields and makes terminology more consistent across the codebase. 'Upload' images have files that were uploaded. 'Copy' images have files that were copied from S3. Transferring is a more general term that refers to uploading or copying. The upload and transfer fields were removed from Layer and upload and transfer fields were consolidated into transfer fields on LayerImage. The upload status is also updated on the client using information from Evaporate. For layers with a mixture of copy and upload images, the status of copying and uploading are displayed independently.

##### Testing
* Submit a layer with an image from S3 and an image from the file system. Check that the status is updated independently for uploading and transferring.
* Try getting the upload to fail as described in #297. You should see an error message from Evaporate in the UI. 
* Try reloading while the upload is in progress. The upload status should be in a failed state. 
* Try reloading when the upload is complete. The upload status should stay in a completed state.

Connects #297
